config/types/ignition.go

```diff
@@ -28,42 +27,31 @@ var (
 	ErrNewVersion = errors.New("incorrect config version (too new)")
 )
 
-type Ignition struct {
-	Version  IgnitionVersion `json:"version,omitempty"  merge:"old"`
-	Config   IgnitionConfig  `json:"config,omitempty"   merge:"new"`
-	Timeouts Timeouts        `json:"timeouts,omitempty" merge:"new"`
-}
-
-type IgnitionConfig struct {
-	Append  []ConfigReference `json:"append,omitempty"`
-	Replace *ConfigReference  `json:"replace,omitempty"`
-}
-
-type ConfigReference struct {
-	Source       Url          `json:"source,omitempty"`
-	Verification Verification `json:"verification,omitempty"`
-}
-
-type IgnitionVersion semver.Version
-
-func (v *IgnitionVersion) UnmarshalJSON(data []byte) error {
-	tv := semver.Version(*v)
-	if err := json.Unmarshal(data, &tv); err != nil {
-		return err
+func (c ConfigReference) ValidateSource() report.Report {
+	r := report.Report{}
+	err := validateURL(c.Source)
+	if err != nil {
+		r.Add(report.Entry{
+			Message: err.Error(),
+			Kind:    report.EntryError,
+		})
 	}
-	*v = IgnitionVersion(tv)
-	return nil
+	return r
 }
 
-func (v IgnitionVersion) MarshalJSON() ([]byte, error) {
-	return semver.Version(v).MarshalJSON()
+func (v IgnitionVersion) Semver() (*semver.Version, error) {
+	return semver.NewVersion(v.Version)
 }
 
 func (v IgnitionVersion) Validate() report.Report {
-	if MaxVersion.Major > v.Major {
+	tv, err := v.Semver()
+	if err != nil {
+		return report.ReportFromError(ErrOldVersion, report.EntryError)
```

Comment:
`ErrOldVersion` seems wrong here; this is invalid, neither too old nor too new.

My previous comment about ignoring an error might be redundant with this one.